### ISSUE TITLE
build error on MacOS/ARM 3.8 

### ISSUE THREAD

**Issue by Anthchirp** [silx-kit/hdf5plugin#115]
I'm trying to build hdf5plugin 2.3.2 on MacOS on ARM64 for conda-forge (conda-forge/hdf5plugin-feedstock#20), and [I see the following build error:](https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=326789&view=logs&j=385a3385-ae8b-54d1-9abe-3c614f74cb73&t=a5f9d343-149e-5621-6ab9-f0278009be04&l=1251)
```
arm64-apple-darwin20.0.0-clang -fno-strict-aliasing -Wsign-compare -Wunreachable-code -DNDEBUG -fwrapv -O3 -Wall -Wstrict-prototypes -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -pipe -fdebug-prefix-map=${SRC_DIR}=/usr/local/src/conda/${PKG_NAME}-${PKG_VERSION} -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol=/usr/local/src/conda-prefix -flto -Wl,-export_dynamic -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O3 -ftree-vectorize -fPIC -fPIE -fstack-protector-strong -O2 -pipe -isystem /Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/include -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/work=/usr/local/src/conda/hdf5plugin-2.3.2 -fdebug-prefix-map=/Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol=/usr/local/src/conda-prefix -D_FORTIFY_SOURCE=2 -isystem /Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/include -mmacosx-version-min=11.0 -DHAVE_LZ4=1 -DHAVE_ZLIB=1 -DHAVE_ZSTD=1 -DH5_USE_18_API -DHAVE_SNAPPY=1 -DSHUFFLE_SSE2_ENABLED=1 -I/Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/include -Isrc/c-blosc/ -Isrc/c-blosc/blosc -Isrc/c-blosc/internal-complibs/lz4-1.9.1 -Isrc/c-blosc/internal-complibs/zlib-1.2.8 -Isrc/c-blosc/internal-complibs/zstd-1.4.1 -Isrc/c-blosc/internal-complibs/zstd-1.4.1/common -Isrc/hdf5-blosc/src/ -Isrc/c-blosc/internal-complibs/snappy-1.1.1 -I/Users/runner/miniforge3/conda-bld/hdf5plugin_1622044221898/_h_env_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehold_placehol/include/python3.8 -c src/c-blosc/blosc/shuffle.c -o build/temp.macosx-11.0-arm64-3.8/src/c-blosc/blosc/shuffle.o -msse2
    clang-11: warning: -Wl,-export_dynamic: 'linker' input unused [-Wunused-command-line-argument]
    clang-11: warning: argument unused during compilation: '-msse2' [-Wunused-command-line-argument]
    src/c-blosc/blosc/shuffle.c:163:5: error: invalid output constraint '=b' in asm
        "=b" (cpuInfo[1]),
        ^
    src/c-blosc/blosc/shuffle.c:188:5: error: invalid output constraint '=a' in asm
        "=a" (eax),
        ^
    2 errors generated.
    error: command 'arm64-apple-darwin20.0.0-clang' failed with exit status 1
```

Is MacOS/ARM supported by hdf5plugin?

**Comment by t20100** [comment]
We haven't tested it on macos/ARM since we don't have access to an ARM64 mac (yet), so it is not supported yet, but we would like to support it.

I made PR #116 which should fix the issue by disabling SSE2 and AVX2 on `arm64` but I can't test it...

**Comment by Anthchirp** [comment]
I made a successful test build using commit f1768e007d782f189663b4fddc33dced7c5228bc (post-116-merge) in conda-forge/hdf5plugin-feedstock#21, so I assume this ticket can be closed.

Do you know when the next release of hdf5plugin will happen?

**Comment by t20100** [comment]
Thanks for the testing!

With M1 support fixed we should make a release soon.

closed by PR #116
